File: Media Ministry Manager/Custom Objects/Types/extensions.vb
Imports System.Runtime.CompilerServices
Imports System.Text.RegularExpressions

Namespace Types.Extensions
	Module StringExtensions
		Private ReadOnly phone As New Regex("(\d{3})(\d{3})(\d{4})")

		<Extension()>
		Public Function FormatPhone(value As String) As String
			Return phone.Replace(value, "({0}) {1}-{2}")
		End Function
	End Module
End Namespace

File: Media Ministry Manager/Custom Objects/Types/extensions.cs
using System.Text.RegularExpressions;

namespace M3App.Types.Extensions
{
    static class StringExtensions
    {
        private readonly static Regex phone = new Regex(@"(\d{3})(\d{3})(\d{4})");

        public static string FormatPhone(this string value)
        {
            return phone.Replace(value, "({0}) {1}-{2}");
        }
    }
}